app/controllers/admin/orders_controller.rb

```diff
@@ -1,6 +1,38 @@
 class Admin::OrdersController < Admin::AdminController
+  before_action :load_order, only: %i(show update)
+
   def index
-    @orders = Order.status_search(params[:status_id]).newest
+    @orders = Order.status_search(params[:status]).newest
     @pagy, @orders = pagy(@orders, items: Settings.item.max_item_10)
   end
+
+  def show; end
+
+  def update
+    new_stt = params[:status].to_sym
+    updated = @order.update_status new_stt if @order.present?
+
+    return render_with_msg :show, :success, t(".successful_update") if updated
+
+    render_with_msg :show, :warning, t(".failed_update")
+  end
+
+  private
+
+  def load_order
+    @order = Order.with_items(params[:id])
```

Comment:
Chị nghĩ ko cần viết scope chỗ này. như bên dưới tường minh hơn nhìu
```suggestion
    @order = Order.includes(:order_items).find_by(id: params[:id])
```